### DISCUSSION TITLE
Integrate with IP intelligence

### DISCUSSION THREAD

**Discussion by mhf-ir** [AltraMayor/gatekeeper#488] [Q&A]
There are bunch of tools like https://github.com/firehol/blocklist-ipsets, https://www.abuseipdb.com/ for track bad behavior IP adresses.

Use case:
 - How tel programmatically that's ip/subnet is white list even when try to reach limitation.
 - How tel programmatically is blacklist even there are no attempt to access the network?
 
 I'm looking for whitelist/blacklist approach.
 
Any database, API or script?

**Comment by andrenth** [comment]
Hello

I have just updated the [Tips for deployments](https://github.com/AltraMayor/gatekeeper/wiki/Tips-for-deployments) page in our Wiki, adding some more detailed information about obtaining IP block lists and feeding them into a Gatekeeper policy. Specifically, the section [Updating LPM tables with with drib and gtctl](https://github.com/AltraMayor/gatekeeper/wiki/Tips-for-deployments#updating-lpm-tables-with-with-drib-and-gtctl) shows how to use the drib tool to manage multiple sources of unwanted IP addresses. 

The wiki page shows an example of how to use drib to load an initial set of prefixes into a Gatekeeper policy, and also how to use it in conjunction with the gtctl tool to perform incremental updates to this set as the blacklists are updated.

Drib also supports whitelist-like functionality via its "priority" mechanism, which is detailed in the project [README](https://github.com/andrenth/drib/blob/master/README.md).

Please let me know if you have more questions.
